metaserver/src/main/scala/scala/meta/languageserver/search/SymbolIndex.scala

```diff
@@ -1,223 +1,39 @@
-import org.langmeta.internal.semanticdb.schema.ResolvedName
 import org.langmeta.internal.semanticdb.{schema => s}
 import org.langmeta.io.AbsolutePath
-import org.langmeta.io.RelativePath
-import org.langmeta.languageserver.InputEnrichments._
-import org.langmeta.semanticdb.Signature
 import org.langmeta.semanticdb.Symbol
 
 /**
  * A high-level wrapper around [[DocumentIndex]] and [[SymbolIndexer]].
  *
  * Can respond to high-level queries like "go to definition" and "find references".
  */
-class SymbolIndex(
-    val symbolIndexer: SymbolIndexer,
-    val documentIndex: DocumentIndex,
-    cwd: AbsolutePath,
-    notifications: Notifications,
-    buffers: Buffers,
-    serverConfig: ServerConfig,
-) extends LazyLogging {
-  private val indexedJars: ConcurrentHashMap[AbsolutePath, Unit] =
-    new ConcurrentHashMap[AbsolutePath, Unit]()
-
-  /** Returns a ResolvedName at the given location */
-  def resolveName(
-      path: AbsolutePath,
-      line: Int,
-      column: Int
-  ): Option[ResolvedName] = {
-    logger.info(s"resolveName at $path:$line:$column")
-    for {
-      document <- documentIndex.getDocument(path.toNIO.toUri)
-      _ = logger.info(s"Found document for $path")
-      input = Input.VirtualFile(document.filename, document.contents)
-      name <- document.names.collectFirst {
-        case name @ ResolvedName(Some(position), symbol, _) if {
-              val range = input.toIndexRange(position.start, position.end)
-              logger.trace(
-                s"${document.filename.replaceFirst(".*/", "")} [${range.pretty}] ${symbol}"
-              )
-              range.contains(line, column)
-            } =>
-          name
-      }
-    } yield name
-  }
-
-  /** Returns a symbol at the given location */
-  def findSymbol(
-      path: AbsolutePath,
-      line: Int,
-      column: Int
-  ): Option[Symbol] = {
-    for {
-      name <- resolveName(path, line, column)
-      symbol = Symbol(name.symbol)
-      _ = logger.info(s"Matching symbol ${symbol}")
-    } yield symbol
-  }
-
-  /** Returns symbol definition data from the index taking into account relevant alternatives */
-  def definitionData(
-      symbol: Symbol
-  ): Option[SymbolData] = {
-    (symbol :: symbol.definitionAlternative)
-      .collectFirst {
-        case symbolIndexer(data) if data.definition.nonEmpty =>
-          logger.info(s"Found definition symbol ${data.symbol}")
-          data
-      }
-  }
-
-  /** Returns symbol references data from the index taking into account relevant alternatives */
-  def referencesData(
-      symbol: Symbol
-  ): List[SymbolData] = {
-    (symbol :: symbol.referenceAlternatives)
-      .collect {
-        case symbolIndexer(data) =>
-          if (data.symbol != symbol.syntax)
-            logger.info(s"Adding alternative references ${data.symbol}")
-          data
-      }
-  }
-
+trait SymbolIndex {
+  def findSymbol(path: AbsolutePath, line: Int, column: Int): Option[Symbol]
```

Comment:
If you rebase on top of master then s/AbsolutePath/Uri

Reply:
Done.